codice/ddf
# [2.19.x] G-9700 expose input validation extension point for downstream validation

#### What does this PR do?
Exposes input enum view's `isValid` method as an ~~extension point~~ (**UPDATE: using plugins now instead**) in order to implement downstream validation logic.
also pulls `getEnumValue` into its own util (InputUtil) to be used by DDF and the downstream project. InputUtil could be further expanded to add more utils for other inputs if needed. 

#### Who is reviewing it? 

@andrewzimmer @mojogitoverhere @Lambeaux @brianfelix 

#### Select relevant component teams: 


@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.


#### How should this be tested?

1. build, profile:install
2. verify no regression when searching, selecting attributes in search form, or viewing results in the inspector 

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #____

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
